buffer.go

```diff
@@ -326,22 +332,43 @@ func (b *buffer) resize(size int) {
 
 func (b *buffer) clone() (clone *buffer) {
 	if b.pool != nil {
-		clone = b.pool.get()
+		clone = b.pool.get(len(b.data))
 	} else {
 		clone = &buffer{refc: 1}
 	}
 	clone.data = append(clone.data, b.data...)
 	return clone
 }
 
+// slice of sync.pools is used for levelled buffering.
+// min size of buffer in each pool is basePoolIncrement*2^i
+//
+// [0] -> 1024
+// [1] -> 2048
+// ...
+// [14] -> 16MB
+// [15] -> 32MB
+const totalPools = 16
```

Comment:
Maybe name something like `numPoolBuckets` or similar? `totalPools` makes it sound like this is the global number of pools in the program, despite being a value local to each buffer pool.